### PR TITLE
Address recent bugs

### PR DESCRIPTION
**Pull request by srbalakr**
## Purpose
This PR address some of the recent bugs
1. Users experience 401 during index build.
   - Refresh token every 5 minutes for open ai calls
2. Suggestion prompt does not work
   - Gpt turbo do not handle lengthy system prompts. Moved document sources to user conversation
   
## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: app/backend/approaches/chatreadretrieveread.py [hunk 1]
```diff
@@ -141,10 +141,10 @@ async def run(self, history: list[dict[str, str]], overrides: dict[str, Any]) ->
             system_message = prompt_override.format(follow_up_questions_prompt=follow_up_questions_prompt)
 
         messages = self.get_messages_from_history(
-            system_message + "\n\nSources:\n" + content,
+            system_message,
             self.chatgpt_model,
             history,
-            history[-1]["user"],
+            history[-1]["user"]+ "\n\nSources:\n" + content,
```
Comment (pamelafox) [inline app/backend/approaches/chatreadretrieveread.py:147]: It seems un-intuitive to me that sources would be part of a user message, so it'd help to add a comment here.

File: scripts/prepdocs.py [hunk 2]
```diff
@@ -35,6 +35,11 @@
 SENTENCE_SEARCH_LIMIT = 100
 SECTION_OVERLAP = 100
 
+OPEN_AI_TOKEN_CACHE = {}
```
Comment (pamelafox) [inline scripts/prepdocs.py:38]: The capitalization makes this seem like a constant, but its a mutable dict, so I think it should just be lowercase. 
https://pep8.org/#constants

File: scripts/prepdocs.py [hunk 3]
```diff
@@ -301,6 +307,12 @@ def remove_from_index(filename):
         # It can take a few seconds for search results to reflect changes, so wait a bit
         time.sleep(2)
 
+# refresh open ai token every 5 minutes
+def refresh_openai_token():
+    if OPEN_AI_TOKEN_CACHE[CACHE_KEY_TOKEN_TYPE] == 'azure_ad' and OPEN_AI_TOKEN_CACHE[CACHE_KEY_CREATED_TIME] + 300 < time.time():
```
Comment (pamelafox) [inline scripts/prepdocs.py:312]: I'm bad with time math, but are you sure this should be less than? I keep thinking it should be >

**Review by pamelafox — CHANGES_REQUESTED** [review]
In the future, I think it'd be nice to send unrelated bug fixes in two different PRs. That makes it easy to patch or revert single fixes as needed. Thanks!

File: app/backend/approaches/chatreadretrieveread.py (hunk 1 above)
Comment (srbalakr) [reply app/backend/approaches/chatreadretrieveread.py]: done

File: scripts/prepdocs.py (hunk 3 above)
Comment (srbalakr) [reply scripts/prepdocs.py]: yes.
assume token created time 00:00
Upto 00:05, we dont refresh token

We refresh token once created time lapses 5 minutes.

**Review by pamelafox — APPROVED** [review]
Okay, I stepped through the time logic with my partner, it does work - I was flipping the times in my head.

**Review by chuwik — APPROVED** [review]

**Pull request merged**